test/pylib/artifact_registry.py

```diff
@@ -23,30 +23,42 @@ class ArtifactRegistry:
     reprsented in the artifact registry. """
 
     def __init__(self) -> None:
-        self.suite_artifacts: Dict[Suite, List[Artifact]] = {}
-        self.exit_artifacts: List[Artifact] = []
+        self.suite_artifacts: Dict[str, List[Artifact]] = {}
```

Comment:
Why change the key to `str`? It was IMO much more readable/understandable when you used `Suite`.

Reply:
because I needed __exit__ key for exit artifacts that do not belong to any suite. In Lua None (nil) can not be a dict key type, so I carried over that habit. I don't have a strong opinion about what is better in Python, using strings or None  + objects, so I'll just do it the way you suggest. I do consider, however, comments which point out bugs in my code more valuable than style comments on bugfixes I made without a reviewer request...

Reply:
done